tracer/src/Datadog.Trace/Debugger/LineProbeResolver.cs

```diff
@@ -72,95 +72,87 @@ private static bool IsThirdPartyCode(Assembly loadedAssembly)
+                    var path = lookup.FindPathThatEndsWith(probeFilePath);
+                    if (path != null)
                     {
-                        return assembly;
+                        sourceFileFullPathFromPdb = path;
+                        assembly = candidateAssembly;
+                        return true;
                     }
                 }
             }
 
-            return null;
+            sourceFileFullPathFromPdb = null;
+            assembly = null;
+            return false;
         }
 
         public LineProbeResolveResult TryResolveLineProbe(ProbeDefinition probe, out BoundLineProbeLocation location)
         {
             location = null;
-            var assembly = FindAssemblyContainingFile(probe.Where.SourceFile);
-            if (assembly == null)
+            if (!TryFindAssemblyContainingFile(probe.Where.SourceFile, out var filePathFromPdb, out var assembly))
             {
-                var message = $"Could not find a source file location for probe {probe.Id}.";
-                Log.Information(message);
-                return new LineProbeResolveResult(LiveProbeResolveStatus.Unbound, message);
+                return new LineProbeResolveResult(LiveProbeResolveStatus.Unbound, "Source file location for probe was not found, possibly because the relevant assembly was not yet loaded.");
             }
 
             using var pdbReader = DatadogPdbReader.CreatePdbReader(assembly);
             if (pdbReader == null)
             {
-                var message = $"Failed to read from PDB for probe ID {probe.Id}";
-                Log.Information(message);
-
-                return new LineProbeResolveResult(LiveProbeResolveStatus.Error, message);
+                return new LineProbeResolveResult(LiveProbeResolveStatus.Error, $"Failed to read from PDB");
             }
 
             if (probe.Where.Lines?.Length != 1 || !int.TryParse(probe.Where.Lines[0], out var lineNum))
             {
-                var message = $"Failed to parse line number for Line Probe {probe.Id}. " +
-                              $"The Lines collection contains {PrintContents(probe.Where.Lines)}.";
-                Log.Error(message);
-
-                return new LineProbeResolveResult(LiveProbeResolveStatus.Error, message);
+                return new LineProbeResolveResult(LiveProbeResolveStatus.Error, $"Failed to parse line number.");
             }
 
-            var method = pdbReader.GetContainingMethodAndOffset(probe.Where.SourceFile, lineNum, column: 0, out var bytecodeOffset);
-            location = new BoundLineProbeLocation(probe, assembly.ManifestModule.ModuleVersionId, method.Token, bytecodeOffset, lineNum);
-            return new LineProbeResolveResult(LiveProbeResolveStatus.Bound);
-
-            string PrintContents<T>(T[] array)
+            var method = pdbReader.GetContainingMethodAndOffset(filePathFromPdb, lineNum, column: null, out var bytecodeOffset);
```

Comment:
The method `TryResolveLineProbe` suggests that upon a failure `false` will be returned. But the call to `GetContainingMethodAndOffset` might throw an exception (for numerous of reasons, NRE, ArgumentOutOfRange and if we will go deeper we might find some more) so maybe surround everything with try/catch blocks so we won't propagate out to the sensitive `LiveDebugger.UpdateProbeInstrumentations` and fail the instrumentation processing?

Reply:
Great catch! Thanks :) Fixed in ba2840632377406ab4aaa4511cc0e9dff51230bc